xpki/csrprov/keyreq.go

```diff
@@ -80,8 +81,8 @@ func (kr *keyRequest) SigAlgo() x509.SignatureAlgorithm {
 // Generate generates a key as specified in the request. Currently,
 // only ECDSA and RSA are supported.
 func (kr *keyRequest) Generate() (crypto.PrivateKey, error) {
-	switch algo := kr.Algo(); algo {
-	case "rsa", "RSA":
+	algo := strings.ToUpper(kr.Algo())
+	if algo == "RSA" || strings.HasPrefix(algo, "RSA-SHA-") {
```

Comment:
please add corresponding tests in https://github.com/go-phorce/dolly/blob/410ea844f53774f751387bf150856d6a16f1def8/xpki/csrprov/keyreq_test.go#L22

Reply:
@sri-vatsan-varadarajan should we just use `strings.HasPrefix(algo, "RSA")`  instead of two conditions since in any case that value is not passed to the next method? Similarly only `strings.HasPrefix(algo, "ECDSA")`  for ECDSA?